client/src/app/core/repositories/assignments/assignment-repository.service.ts

```diff
@@ -59,20 +77,14 @@ export class AssignmentRepositoryService extends BaseIsAgendaItemAndListOfSpeake
     /**
      * Adds/removes another user to/from the candidates list of an assignment
      *
-     * @param user A ViewUser
-     * @param assignment The assignment to add the candidate to
-     * @param adding optional boolean to force an add (true)/ remove (false)
-     * of the candidate. Else, the candidate will be added if not on the list,
-     * and removed if on the list
+     * @param userId The id of a view-user
+     * @param assignment The assignment to add or remove the candidate to
      */
-    public async changeCandidate(userId: number, assignment: ViewAssignment, adding?: boolean): Promise<void> {
-        const data = { user: userId };
-        if (assignment.candidates.some(candidate => candidate.id === userId) && adding !== true) {
-            throw new Error('TODO');
-            // await this.httpService.delete(this.restPath + assignment.id + this.candidatureOtherPath, data);
-        } else if (adding !== false) {
-            throw new Error('TODO');
-            // await this.httpService.post(this.restPath + assignment.id + this.candidatureOtherPath, data);
+    public async changeCandidate(userId: number, assignment: ViewAssignment): Promise<void | Identifiable> {
```

Comment:
This can be refactored. Please remove this manifold function and move the logic into the components.